js/ui/windowManager.js

```diff
@@ -1776,14 +1776,20 @@ var WindowManager = new Lang.Class({
                                onCompleteParams: [shellwm, actor],
                                onOverwrite: onOverwrite,
                                onOverwriteScope: this,
-                               onOverwriteParams: [shellwm, actor]
+                               onOverwriteParams: [shellwm, actor],
+                               onUpdate: this._clipActor,
+                               onUpdateParams: ["top", endY, actor],
                              });
         } else {
             let endX;
-            if (actor.x <= monitor.x)
+            let origin;
+            if (actor.x <= monitor.x) {
                 endX = monitor.x - actor.width;
-            else
+                origin = "left";
```

Comment:
Instead of using strings for determining directions, I think it'd be better to use constant pre-declared values. You can even use `ClutterContentGravity` for this.

Reply:
I've used the ClutterContentGravity as you suggested